Refactor generator to support future features (#7214)
rathena/rathena: doc/map_server_generator.md

# Map Server Generator

This is a tool to generate files that are hard to manually create.

> Right now, the only thing you can make are client navigation files. 

Reviewer: ```suggestion
Right now, the only feature is creating the client navigation files.
```

> Right now, the only thing feature is creating the client navigation files.

Reviewer 2: ```suggestion
Only navigation files can be generated at this time.
```

## How to run
### Linux
Run `make tools`.
This creates a new binary called `map-server-generator`.

> You can run this with `./map-server-generator`

Reviewer: ```suggestion
It can be ran with: `./map-server-generator`
```
